### PR TITLE
Switch from Lit-managed rendering to direct DOM manipulation in html-block to avoid conflicts with MathJax

### PR DESCRIPTION
**Pull request by grant-cleary**
A while back, when I introduced the embeds rendering feature, I tried to get rendering with Lit working at the same time. I thought I had succeeded, but alas, not so much. When we run the MathJax renderer, it wipes out the contents of the render container. This also wipes out Lit's marker comments, breaking subsequent renders. This went unnoticed at the time because it requires 1) an update of the `html` property (fresh renders work just fine), and 2) math to be present in the original HTML.

Anyways, I've gone back and forth with a bunch of ideas, but I haven't really found a Lit-native solution that isn't either kind of sketchy (like inferring the existence of Lit's markers), or would just be hiding direct DOM manipulation under a veneer of Lit rendering.

I'm definitely open to ideas from others (I won't discount that I may be missing a good solution here), but I'm personally out of ideas 😅. So for now, this PR just switches back to good old direct DOM manipulation and calls it a day.

### CODE REVIEW

**Review by dlockhart — APPROVED** [review]
Nothing profound is coming to mind!

File: components/html-block/html-block.js [hunk 1]
```diff
@@ -238,16 +236,14 @@ class HtmlBlock extends LitElement {
...
-			</div>
+			<div class="${classMap(renderContainerClasses)}" ${ref(this._renderContainerRef)}></div>
 			${this.noDeferredRendering ? html`<slot @slotchange="${this._handleSlotChange}"></slot>` : ''}
 		`;
 	}
 
 	async updated(changedProperties) {
 		super.updated(changedProperties);
-		if ((changedProperties.has('embeds') || changedProperties.has('_context')) && this.html !== undefined && this.html !== null && !this.noDeferredRendering) {
+		if ((changedProperties.has('html') || changedProperties.has('_context')) && this.html !== undefined && this.html !== null && !this.noDeferredRendering) {
```
Comment (grant-cleary) [inline components/html-block/html-block.js:246]: I think an argument could be made that we should be updating this on every property change. Actually, I think I will make that change. MathJax's styling can change if we switch `compact`, which is the only property here that _doesn't_ warrant a re-run of the renderers.

File: components/html-block/html-block.js [hunk 2]
```diff
@@ -208,6 +208,7 @@ class HtmlBlock extends LitElement {
 
 		this._renderersProcessedResolve = undefined;
 		this._renderersProcessedPromise = new Promise(resolve => this._renderersProcessedResolve = resolve);
+		this._renderContainerRef = createRef();
```
Comment (grant-cleary) [inline components/html-block/html-block.js:211]: This isn't strictly needed, but I do think it's a more stable way to fetch this reference.

**Comment by grant-cleary** [comment]
Ah right, this still has that problem with the timing on the update. Sigh. I don't really understand why I didn't have this issue before the HTML embeds stuff came into play... I'm going to have to take another look at that, I think.

**Comment by grant-cleary** [comment]
Ok, the new vdiff failure isn't related to the html-block, so the mixin was in fact the kicker. The new vdiff image definitely looks more correct than the golden, so I'm going to go ahead and merge that in.

**Comment by grant-cleary** [comment]
Alright, never mind, looks like that test might be a bit flaky?

**Pull request merged**